liquidctl/driver/commander_pro.py

```diff
@@ -0,0 +1,514 @@
+        It will look up the name of the led channel given and return a list of the real led device number
+        """
+        channel = channel.lower()
+        if channel == 'led':
+            return [i for i in range(len(self._led_names))]
+        elif channel in self._led_names:
+            return [self._led_names.index(channel)]
+        else:
+            raise ValueError(f'unknown channel, should be one of: {_quoted("led", *self._led_names)}')
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set fan or fans to a fixed speed duty.
+
+        Valid channel values are 'fanN', where N >= 1 is the fan number, and
+        'fan', to simultaneously configure all fans.  Unconfigured fan channels
+        may default to 100% duty.
+
+        Different commands for sending fixed percent (0x23) and fixed rpm (0x24)
+        Probably want to use fixed percent for this untill the rpm flag is enabled.
+        Can only send one fan command at a time, if fan mode is unset will need to send 6?
+        messages (or 1 per enabled fan)
+        """
+
+        if self._fan_count == 0:
+            raise NotSupportedByDevice()
+
+        duty = clamp(duty, 0, 100)
+        fan_channels = self._get_hw_fan_channels(channel)
+        fan_modes = self._data.load('fan_modes', default=[0]*self._fan_count)
+
+        for fan in fan_channels:
+            mode = fan_modes[fan]
+            if mode == _FAN_MODE_DC or mode == _FAN_MODE_PWM:
+                self._send_command(_CMD_SET_FAN_DUTY,[fan, duty])
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set fan or fans to follow a speed duty profile.
+
+        Valid channel values are 'fanN', where N >= 1 is the fan number, and
+        'fan', to simultaneously configure all fans.  Unconfigured fan channels
+        may default to 100% duty.
+
+        Up to six (temperature, duty) pairs can be supplied in `profile`,
+        with temperatures in Celsius and duty values in percentage.  The last
+        point should set the fan to 100% duty cycle, or be omitted; in the
+        latter case the fan will be set to max out at 60°C.
+
+
+        there are 6 points on the fan curve, temp, rpm pairs
+        Send one message per fan that is being set
+        """
+
+        # send fan num, temp sensor, check to make sure it is actually enabled, and do not let the user send external sensor
+        # 6 2-byte big endian temps (celsius * 100), then 6 2-byte big endian rpms
+        # need to figure out how to find out what the max rpm is for the given fan
+
+        if self._fan_count == 0:
+            raise NotSupportedByDevice()
+
+        profile = list(profile)
+        profile = _prepare_profile(profile)
+
+        # fan_type = kwargs['fan_type'] # need to make sure this is set
+        temp_sensor = kwargs.get('temp_sensor', 1) # need to make sure this is set and in range 1-4 or ext
```

Comment:
Same issue as mentioned bellow (make this an explicit parameter of the function).